File: snippets/python/00-connection/03-session-options.py
# import the shared client:  from elo_playground import connect
# topic:    List session options
# category: Connection & session
# id:       connection.session-options

import os
from elo_playground import connect

# --- local ELO test box (override with ELOPG_* env vars or a .env) ---
ELO_BASE_URL = os.getenv("ELOPG_ELO_BASE_URL", "http://localhost:9090/ix-Repository1") # ELOPG_DEFAULT:base_url
ELO_USER = os.getenv("ELOPG_ELO_USER", "Administrator") # ELOPG_DEFAULT:user
ELO_PASS = os.getenv("ELOPG_ELO_PASSWORD", "elo") # ELOPG_DEFAULT:password

elo = connect(base_url=ELO_BASE_URL, user=ELO_USER, password=ELO_PASS)

res = elo.call("getSessionOptions", {})
options = res.get("options", res if isinstance(res, list) else [])

print(f"{len(options)} session options")
for opt in options[:10]:                       # just the first few
    print(f"  {opt['key']} = {opt['value']}")

File: snippets/node/00-connection/03-session-options.mjs
// import the shared client:  import { connect } from "elo-playground";
// topic:    List session options
// category: Connection & session
// id:       connection.session-options

import { connect } from "elo-playground";

// --- local ELO test box (override with ELOPG_* env vars or a .env) ---
const ELO_BASE_URL = process.env.ELOPG_ELO_BASE_URL || "http://localhost:9090/ix-Repository1"; // ELOPG_DEFAULT:base_url
const ELO_USER = process.env.ELOPG_ELO_USER || "Administrator"; // ELOPG_DEFAULT:user
const ELO_PASS = process.env.ELOPG_ELO_PASSWORD || "elo"; // ELOPG_DEFAULT:password

const elo = await connect({ baseUrl: ELO_BASE_URL, user: ELO_USER, password: ELO_PASS });

const res = await elo.call("getSessionOptions", {});
const options = res.options || (Array.isArray(res) ? res : []);

console.log(`${options.length} session options`);
for (const opt of options.slice(0, 10)) {
  console.log(`  ${opt.key} = ${opt.value}`);
}
